src/main/java/it/polimi/telco/model/ServicePackage.java

```diff
@@ -0,0 +1,39 @@
+package it.polimi.telco.model;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+
+@Entity
+@Table(name = "service_package", schema = "telco")
+@NamedQuery(name = "ServicePackage.getServicePackageById", query = "SELECT sp FROM ServicePackage sp  WHERE sp.id = ?1 ")
+public class ServicePackage {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    @OneToMany
+    private List<TelcoService> services;
+
```

Comment:
_"Also, for all the optional products offered by the company, for each service package only a subset of them is available, right? So the user can select optional products for a service package only from this subset, or am I missing something?
a. Yes, the interpretation is correct"_

We also have to associate a service package with one or more optional products (the user can select 0 or more of those during the subscription)